.github/workflows/lint.yml

```diff
@@ -26,7 +26,10 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
     - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
     - name: golangci-lint
       uses: reviewdog/action-golangci-lint@v2
       with:
+        fail_on_error: true
```

Comment:
This shouldn't be added. The ReviewDog check currently fails only if there is a linting issue with code changed in the PR, not anywhere else in the code, which is how it should be. See https://github.com/runatlantis/atlantis/actions/runs/6070040971 for an example.

Reply:
That's the goal. I don't think a linter should fail only with files that the PR touched, if something comes up on a PR it should be blocked until the error is fixed.

I think we should leave this decision to the maintainers.

Reply:
The linter check has been failing for ages until I fixed it in PR #3645. The linters used by golangci-lint are updated all the time, so definitely a PR shouldn't be blocked because of a new linting issue in code that is not touched by the PR.

Reply:
IMHO that's how it got to this state. I understand your point but this is speed vs stability. How can we develop properly and run `make lint` to see 200 errors? Did we caused them or were they already there, locally we don't get this information. I believe the codebase needs to be always clean.

Reply:
removed to unblck this.